File: src/leetcode/sum-of-unique-elements/sum-of-unique-elements.js
/**
 * Leetcode #1748
 * Difficulty: Easy
 * https://leetcode.com/problems/sum-of-unique-elements/
 */
/**
 * @param {number[]} nums
 * @return {number}
 */
export const sumOfUnique = nums => {
  const map = new Map();

  for (const num of nums) {
    map.set(num, (map.get(num) ?? 0) + 1);
  }

  let sum = 0;

  for (const num of nums) {
    const count = map.get(num);

    if (count === 1) {
      sum += num;
    }
  }

  return sum;
};

File: src/leetcode/sum-of-unique-elements/sum-of-unique-elements.rs
impl Solution {
    pub fn sum_of_unique(nums: Vec<i32>) -> i32 {
        let mut map = std::collections::HashMap::new();

        for num in nums.iter() {
            *map.entry(num).or_insert(0) += 1;
        }

        nums
            .iter()
            .filter(|num| *map.get(num).unwrap() == 1)
            .sum::<i32>() as i32
    }
}
